src/middlewared/middlewared/plugins/smart.py

```diff
@@ -4,6 +4,7 @@
 import re
```

Comment:
Something I've noticed in this file. On line 115 you have this
```
            if not test_key in data:
                break
```
This should be written as `if test_key not in data`